src/go/k8s/pkg/console/configmap.go

```diff
@@ -0,0 +1,310 @@
+	if cm.consoleobj.Status.ConfigMapRef != nil {
+		return nil
+	}
+
+	// Create new ConfigMap
+	// If reconciliation fails, a new ConfigMap will be created again
+	// But unused ConfigMaps should be deleted at the beginning of reconciliation via DeleteUnused()
+
+	secret := v1.Secret{}
+	if err := cm.Get(ctx, KafkaSASecretKey(cm.consoleobj), &secret); err != nil {
+		return err
+	}
+	username := string(secret.Data[corev1.BasicAuthUsernameKey])
+	password := string(secret.Data[corev1.BasicAuthPasswordKey])
+
+	config, err := cm.generateConsoleConfig(ctx, username, password)
+	if err != nil {
+		return err
+	}
+
+	// Create new ConfigMap instead of updating existing so Deployment will trigger a reconcile
+	immutable := true
+	obj := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: cm.consoleobj.GetName() + "-",
+			Namespace:    cm.consoleobj.GetNamespace(),
+			Labels:       labels.ForConsole(cm.consoleobj),
+		},
+		Data: map[string]string{
+			"config.yaml": config,
+		},
+		Immutable: &immutable,
+	}
+
+	if err := controllerutil.SetControllerReference(cm.consoleobj, obj, cm.scheme); err != nil {
+		return err
+	}
+	if err := cm.Create(ctx, obj); err != nil {
+		return fmt.Errorf("creating Console configmap: %w", err)
+	}
+
+	// This will get updated in the controller main reconcile function
+	// Other Resources may set Console status if they are also watching GenerationMatchesObserved()
+	cm.consoleobj.Status.ConfigMapRef = &corev1.ObjectReference{Namespace: obj.GetNamespace(), Name: obj.GetName()}
+
+	return nil
+}
+
+// Key implements Resource interface
+func (cm *ConfigMap) Key() types.NamespacedName {
+	return types.NamespacedName{Name: cm.consoleobj.GetName(), Namespace: cm.consoleobj.GetNamespace()}
+}
+
+// generateConsoleConfig returns the actual config passed to Console.
+// This should match the fields at https://github.com/redpanda-data/console/blob/master/docs/config/console.yaml
+// We are copying the fields instead of importing them because (1) they don't have json tags (2) some fields aren't ideal for K8s (e.g. TLS certs shouldn't be file paths but Secret reference)
+func (cm *ConfigMap) generateConsoleConfig(ctx context.Context, username, password string) (string, error) {
+	consoleConfig := &ConsoleConfig{
+		MetricsNamespace: cm.consoleobj.Spec.MetricsNamespace,
+		ServeFrontend:    cm.consoleobj.Spec.ServeFrontend,
+		Server:           cm.consoleobj.Spec.Server,
+		Kafka:            cm.genKafka(username, password),
+	}
+
+	connectConfig, err := cm.genConnect(ctx)
+	if err != nil {
+		return "", err
+	}
+	consoleConfig.Connect = *connectConfig
+
+	config, err := yaml.Marshal(consoleConfig)
+	if err != nil {
+		return "", err
+	}
+	return string(config), nil
+}
+
+var (
+	// Currently issuing TLS certs through LetsEncrypt
+	// SchemaRegistry.TLS.NodeSecretRef is set to Secret without CaFile "ca.crt"
+	// This flag overrides using NodeSecretRef to mount CaFile
+	UsePublicCerts = true
```

Comment:
Can this be dynamically computed? I guess this does not work in local dev with self-signed certs, right?

Can be also done in a followup

Reply:
I think we can use if `IssuerRef` is set for the TLS. But I will confirm, I will update in follow up PR

Reply:
On another thought, I think we can't use IssuerRef. I provisioned a v1 Cluster and the SchemaRegistry IssuerRef=nil and NodeSecretRef is set but it is using LetsEncrypt.